microcloud/mdns/mdns.go

```diff
@@ -48,3 +50,44 @@ func dnsTXTSlice(list []byte) []string {
 
 	return parts
 }
+
+// checkIPStatus checks if the interface is up, has multicast, and supports IPv4/IPv6.
+func checkIPStatus(iface string) (ipv4OK bool, ipv6OK bool, err error) {
+	netInterface, err := net.InterfaceByName(iface)
+	if err != nil {
+		return false, false, err
+	}
+
+	if netInterface.Flags&net.FlagUp != net.FlagUp {
+		fmt.Printf("Interface %q is down\n", iface)
```

Comment:
Why are you printing output here?

Reply:
exactly. I removed the printfs, but should I keep them as `logger.Warnf` ?